### DISCUSSION TITLE
使用 jsDelivr 免费加速 GitHub Pages 博客的静态资源（二） — 码志

### DISCUSSION THREAD

**Discussion by mzlogin** [mzlogin/blog-comments#216] [Announcements]
https://mazhuang.org/2020/10/07/cdn-for-github-pages-2/ 

 之前写过一篇 使用 jsDelivr 免费加速 GitHub Pages 博客的静态资源，在那之后，又陆续想到并实施了几点利用 jsDelivr 进一步加速静态资源加载的措施，新起一篇作为记录和分享。

**Comment by bran-nie** [comment]
这里呀，我发现有个问题：
当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。

**Comment by bran-nie** [comment]
> 这里呀，我发现有个问题：
> 当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。

后面发现，是本地浏览器的缓存

**Comment by mzlogin** [comment]
> > 这里呀，我发现有个问题：
> > 当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。
> 
> 后面发现，是本地浏览器的缓存

👍 好的，感谢反馈和更新，那应该还好。

**Comment by bran-nie** [comment]
> @mzlogin
> > > 这里呀，我发现有个问题：
> > > 当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。
> > 
> > 后面发现，是本地浏览器的缓存
> 
> 👍 好的，感谢反馈和更新，那应该还好。

后续我发现呀，这个 search_json，不走缓存比较好。
1. 因为用户很少会强制刷新网站，而 search_json 被缓存后，新的博客，就没有在 json 中了。至于什么时候会有，至少3天都没更新(因为你的新博客 27 号发布，30 号，search_json 文件还没更新。
2. CDN 本意是静态资源加速，而这个 json 文件，会经常变动。

---
补充：当使用自定义域名后，请求资源相比原本的`github.io`快多了。此外，json 文件也是通过搜索插件配置的，插件是发的异步请求，也不会影响首屏加载的～

**Comment by mzlogin** [comment]
> @bran-nie
> 这里呀，我发现有个问题：
> 当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。

我打算给 search_data.json 文件的请求链接后面加个版本号，值就是 build 时的时间戳，这样当有新的 build 时，版本号变动自动刷新缓存，没有时就走缓存。

**Comment by bran-nie** [comment]
> > @bran-nie
> > 这里呀，我发现有个问题：
> > 当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。
> 
> 我打算给 search_data.json 文件的请求链接后面加个版本号，值就是 build 时的时间戳，这样当有新的 build 时，版本号变动自动刷新缓存，没有时就走缓存。

这个我不熟悉，现在没能想到，如何在 build 时，能将时间戳添加到 json 文件的请求链接上。

设置一个变量？当执行 build 的时候，就重新赋值给这个变量，随后，json 文件的请求那里，就引用？

**Comment by mzlogin** [comment]
> @bran-nie
> > > @bran-nie
> > > 这里呀，我发现有个问题：
> > > 当 search_data.json 被 cdn 缓存后，即便后续 built 分支更新了，但页面加载的 json 文件还是旧的，因为页面请求的 json，是 cdn 上面的缓存，而不是 built 分支最新的。
> > 
> > 我打算给 search_data.json 文件的请求链接后面加个版本号，值就是 build 时的时间戳，这样当有新的 build 时，版本号变动自动刷新缓存，没有时就走缓存。
> 
> 这个我不熟悉，现在没能想到，如何在 build 时，能将时间戳添加到 json 文件的请求链接上。
> 
> 设置一个变量？当执行 build 的时候，就重新赋值给这个变量，随后，json 文件的请求那里，就引用？

可以用 Liquid 语法来实现，参考我的更改：<https://github.com/mzlogin/mzlogin.github.io/commit/268f60d8c6f6ec694f67d52914630d138e22eb4e>

Liquid 语法参考：<https://liquid.bootcss.com/filters/date/>

**Comment by bran-nie** [comment]
谢谢～ 我已经试过并添加了。

这个`now`是环境变量吗？相关还有其他的么，我没能找到相关的文档，你在哪里看到的呢

**Comment by mzlogin** [comment]
> @bran-nie
> 谢谢～ 我已经试过并添加了。
> 
> 这个`now`是环境变量吗？相关还有其他的么，我没能找到相关的文档，你在哪里看到的呢
> 

我就是在上面发的 Liquid 文档里看到的，目前也不知道是否还有其它的。
